File: DAA.Models/Settings/TaskTemplate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DAA.Models.Settings
{
    public class TaskTemplateModel
    {
        public int TaskTemplateId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string RelatedContentUrl { get; set; }
        public ICollection<int> ProcessesStepId { get; set; }
    }

    public class TaskTemplateDisplayModel
    {
        public int TaskTemplateId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string RelatedContentUrl { get; set; }
    }

    public class TaskTemplateCreateModel
    {
        public string Title { get; set; }
        public string Description { get; set; }
        public string RelatedContentUrl { get; set; }
        public ICollection<int> ProcessesStepId { get; set; }
    }

    public class TaskTemplateUpdateModel : TaskTemplateCreateModel
    {
        public int TaskTemplateId { get; set; }
    }
}

File: DAA.Web.Intranet/client/src/models/taskTemplate.ts
export class TaskTemplateDisplayModel implements ITaskTemplate {
	constructor(obj: ITaskTemplate = {} as ITaskTemplate) {
		this.taskTemplateId = obj.taskTemplateId;
		this.title = obj.title;
		this.description = obj.description;
		this.relatedContentUrl = obj.relatedContentUrl;
	}

	taskTemplateId: number;
	title: string;
	description: string;
	relatedContentUrl: string;
}

export interface ITaskTemplate {
	taskTemplateId: number;
	title: string;
	processesStepId?: Array<number>;
	description: string;
	relatedContentUrl: string;
}

export class TaskTemplateCreateModel {
	title: string = '';
	description: string = '';
	relatedContentUrl: string = '';
	processesStepId: Array<number> = [];
}

export class TaskTemplateUpdateModel extends TaskTemplateDisplayModel {
	constructor(obj: ITaskTemplate = {} as ITaskTemplate) {
		super(obj);
		this.processesStepId = obj.processesStepId
	}
	processesStepId?: number[];
}